proxy/context.go

```diff
@@ -206,6 +210,38 @@ func (c *context) Metrics() filters.Metrics            { return c.metrics }
 func (c *context) Tracer() opentracing.Tracer          { return c.tracer }
 func (c *context) ParentSpan() opentracing.Span        { return c.parentSpan }
 
+func (c *context) Logger() *log.Entry {
+	traceId := c.getTraceID()
+	return log.WithFields(log.Fields{"trace_id": traceId})
+}
+
+func (c *context) getTraceID() string {
+	if c == nil {
+		return "(nil context)"
+	}
+	if c.request == nil {
+		return "(nil request)"
+	}
```

Comment:
Is it possible for context and request to be nil actually?

Reply:
I've seen such case in `make cicheck`. Will look deeped whether it's applicable for real life.

Reply:
Okay, context and request both cannot be nil both for `proxy.context` and `script.luaContext`. However, `request == nil` is possible for `filtertest.Context` so I will keep this check only there.